psydac/linalg/solvers.py

```diff
@@ -1718,3 +1690,263 @@ def solve(self, b, out=None):
 
     def dot(self, b, out=None):
         return self.solve(b, out=out)
+
+#===============================================================================
+class GMRES(InverseLinearOperator):
+    """
+    A LinearOperator subclass. Objects of this class are meant to be created using :func:~`solvers.inverse`.
+
+    The .dot (and also the .solve) function are based on the 
+    Generalized minimum residual algorithm for solving linear system Ax=b.
```

Comment:
There are two small errors in the name:

- minimum -> minimal
- "generalized" should not be capitalized